### PR TITLE
Makefile

### LINKED ISSUE #1: `make clean` fails on TARGET
doesnt work for `make clean` here:

``` Makefile
TARGETS=$(patsubst dates/%.yaml, dates/%/ALL, $(wildcard dates/*.yaml))
```

GNU Make 4.1

### PR DESCRIPTION
**Pull request by vv01f**
- clean didnt work for me, TARGET with "ALL" did not list files for `rm` - fixes #1 for me
- added error reporting in case interpreter or script is missing
- enabled adding new scripts for other interpreters later, e.g. python implementation of generator; so users do not need to install ruby

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+	rm -rf ${CLEAN_TARGETS}
 
 # Keep Makefiles
-.SECONDARY: $(patsubst dates/%/ALL, dates/%/Makefile, $(TARGETS))
+.SECONDARY: $(patsubst dates/%/ALL, dates/%/Makefile, $(MAKE_TARGETS))
 
 dates/%/Makefile: dates/%.yaml $(wildcard decks/*.yaml)
 	mkdir -p $(shell dirname $@)
-	./generate.rb $< $(shell dirname $@)
+	$(GENERATOR) $< $(shell dirname $@) || printf "\033[1;31minterpreter or script missing\033[0m\n"
```
Comment (Mic92) [inline Makefile]: Is it intended to ignore the error here?

Comment (s72785) [reply Makefile]: it was itended to make the user aware of the reason as it might be used of people not familiar with Make, LaTeX nor Ruby or alike.

Comment (Mic92) [reply Makefile]: The version above would generate false positives, if ruby is installed but the command failed.

``` patch
+   RUBY := ruby

    dates/%/Makefile: dates/%.yaml $(wildcard decks/*.yaml)
         mkdir -p $(shell dirname $@)
-       ./generate.rb $< $(shell dirname $@)
+       command -v $(RUBY) || (printf "\033[1;31Please install Ruby!\033[0m\n"; exit 1)
+       $(RUBY) ./generate.rb $< $(shell dirname $@)

```

Comment (s72785) [reply Makefile]: when does it fail while ruby is installed? (I do not know; apart the bin is not where it should be or the file aint executable)

Comment (Mic92) [reply Makefile]: Let's say the ruby script exit due an Exception or whatever reason

```
target:
    ruby -e "raise 'Oh no!'" || printf "\033[1;31minterpreter or script missing\033[0m\n"
```

``` sh
make: Entering directory '/etc/nixos/nixpkgs'
ruby -e "raise 'Oh no!'" || printf "\033[1;31minterpreter or script missing\033[0m\n"
-e:1:in `<main>': Oh no! (RuntimeError)
interpreter or script missing
make: Leaving directory '/etc/nixos/nixpkgs'
```

Comment (farao) [reply Makefile]: @vv01f Do you agree that this is a good change and can you update your pull request?

Comment (vv01f) [reply Makefile]: @farao ack, will update

Comment (vv01f) [reply Makefile]: updated and dropped coloing as it didn't work any more on my machine and also ate first char of the string

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**